doc/developing_in_windows_msys.md

```diff
@@ -0,0 +1,140 @@
+
+1. Rename the `msys64` folder to `msys64.bak`.
+
+2. Use the installer to reinstall MSYS2 to `msys64`.
+
+3. Copy packages from `msys64.bak/var/cache/pacman/pkg/` to the new installation to save download time and bandwidth.
+
+4. Use `pacman` from within the MSYS environment to install and update packages.
+
+5. After successfully building a project, it is safe to delete `msys64.bak`
+
+## The MING64 Environment
+
+Start by setting up the MING64 environment.  The following commands should be run within the MSYS environment.  `pacman -S` is used to install packages.  The `--needed` argument prevents packages from being reinstalled.
+
+    pacman -S --needed base-devel mingw-w64-x86_64-toolchain
+    pacman -S git mingw-w64-x86_64-cmake mingw-w64-x86_64-ninja \
+        mingw-w64-x86_64-gtest mingw-w64-x86_64-giflib \
+        mingw-w64-x86_64-libpng mingw-w64-x86_64-libjpeg-turbo 
+
+If any packages are found to be missing later, you may search for them with:
+
+    pacman -Ss [search-term]
+
+When searching for packages, results are in the form: `[group]/package-name`.  For example:
+
+    mingw32/mingw-w64-i686-cmake
+    mingw64/mingw-w64-x86_64-cmake
+    ucrt64/mingw-w64-ucrt-x86_64-cmake
+    clang64/mingw-w64-clang-x86_64-cmake
+    msys/cmake
+
+For tools that need to be aware of the compiler to function, install the package that corresponds with the specific build-environment you plan to use.  For `cmake`, install the `mingw64` version.  The generic `msys/cmake` will not function correctly because it will not find the compiler.  For other tools, the generic `msys` version is adequate, like `msys/git`.
+
+To remove packages, use:
+
+    pacman -Rsc [package-name]
+
+## Other Environments
+
+MSYS2 provides multiple development [environments](https://www.msys2.org/docs/environments/), which are layers on top of the MSYS environment.  They target slightly different platforms, runtime libraries, and compiler toolchains.  Remember to always make package changes from within the MSYS environment.  Then `exit` all MSYS2 environments and restart the desired build environment.
+
+* **MINGW32:**  To develop for 32-bit Windows, replace `x86_64` with `i686` in the package names. 
+
+* **UCRT64:**  The Universal C Runtime (UCRT) is used by recent versions of Microsoft Visual Studio.  It ships by default with Windows 10.  For older versions of Windows, it must be provided with the application or installed by the user.  To use the UCRT64 environment, follow the above instructions, but use the `mingw-w64-ucrt-x86_64` prefix in the package names.
+
+* **CLANG64:** Unfortunately, the `gimp` packages are not available for the CLANG64 environment.  However, `libjxl` will otherwise build in this environment if the appropriate packages are installed.  Packages are prefixed with `mingw-w64-ucrt64-x86_64`.
```

Comment:
```suggestion
* **CLANG64:** Unfortunately, the `gimp` packages are not available for the CLANG64 environment.  However, `libjxl` will otherwise build in this environment if the appropriate packages are installed.  Packages are prefixed with `mingw-w64-clang64-x86_64`.
```

Reply:
Thanks.  Will fix.